x-pack/plugins/transform/public/app/hooks/use_transform_config_data.ts

```diff
@@ -147,100 +145,107 @@ export const useTransformConfigData = (
     tableItems,
   } = dataGrid;
 
-  const getPreviewData = async () => {
-    if (!validationStatus.isValid) {
+  const previewRequest = useMemo(
+    () =>
+      getPreviewTransformRequestBody(
+        dataView,
+        query,
+        requestPayload,
+        combinedRuntimeMappings,
+        timeRangeMs
+      ),
+    [dataView, query, requestPayload, combinedRuntimeMappings, timeRangeMs]
+  );
+
+  const {
+    error: previewError,
+    data: previewData,
+    isError,
+    isLoading,
+  } = useGetTransformsPreview(previewRequest, validationStatus.isValid);
+
+  useEffect(() => {
```

Comment:
I reckon we should avoid such massive effects. It seems like it can be broken down in independent variables